### PR TITLE
L2-439: More deterministic e2e screenshot

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

Make e2e test after registration screenshot more deterministic.

# Changes

* Wait for the toast error and remove it before taking screenshot.

# Tests

The changes are in the e2e test.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
@bitdivine @peterpeterparker please review

File: e2e-tests/specs/home.e2e.ts [hunk 1]
```diff
@@ -26,19 +26,31 @@ describe("landing page", () => {
...
-    // remove spinner to make screenshots deterministic
-    const spinner = await browser.$('main section svg');
-    await spinner.waitForExist();
-    await browser.execute(() => document.querySelector('main section svg').remove());
-    await waitForImages(browser);
-
-    await browser["screenshot"]("home-page");
+    try {
+      // remove spinner to make screenshots deterministic
+      const spinner = await browser.$("main section svg");
```
Comment (peterpeterparker) [inline e2e-tests/specs/home.e2e.ts:37]: use `data-id` selector

File: e2e-tests/specs/home.e2e.ts [hunk 2]
```diff
@@ -26,19 +26,31 @@ describe("landing page", () => {
...
+      await spinner.waitForExist();
+      await browser.execute(() =>
+        document.querySelector("main section svg").remove()
+      );
+      await waitForImages(browser);
+      // remove toast error to make screenshots deterministic
+      const errorToast = await browser.$(".toast");
+      await errorToast.waitForExist({ timeout: 10_000 });
+      await browser.execute(() => document.querySelector(".toast").remove());
```
Comment (peterpeterparker) [inline e2e-tests/specs/home.e2e.ts:46]: should you add a "wait for .toast not exists"? it's super fast, like milli-seconds, but remove in the dom !== effectively re-rendered by the browser therefore it is still possible in terrible edge case that the screenshot is still taken with the toast if I get it correctly

**Review by peterpeterparker — CHANGES_REQUESTED** [review]
Coolio. A small note and a question

File: e2e-tests/specs/home.e2e.ts (hunk 1 above)
Comment (lmuntanerDfinity) [reply e2e-tests/specs/home.e2e.ts:37]: ok

File: e2e-tests/specs/home.e2e.ts (hunk 2 above)
Comment (lmuntanerDfinity) [reply e2e-tests/specs/home.e2e.ts:46]: good point, it makes sense to check that it's not there.

Comment (lmuntanerDfinity) [reply e2e-tests/specs/home.e2e.ts:46]: done

File: e2e-tests/specs/home.e2e.ts [hunk 3]
```diff
@@ -26,19 +26,32 @@ describe("landing page", () => {
...
-    await browser.execute(() => document.querySelector('main section svg').remove());
-    await waitForImages(browser);
-
-    await browser["screenshot"]("home-page");
+    try {
+      // remove spinner to make screenshots deterministic
+      const spinner = await browser.$('[data-tid="spinner"]');
+      await spinner.waitForExist();
+      await browser.execute(() =>
+        document.querySelector("main section svg").remove()
```
Comment (peterpeterparker) [inline e2e-tests/specs/home.e2e.ts:40]: you can use `[data-tid="spinner"]` here too 😉

Comment (lmuntanerDfinity) [reply e2e-tests/specs/home.e2e.ts:40]: oops, yes
